server/src/main/java/org/elasticsearch/index/mapper/LeafRuntimeField.java

```diff
@@ -20,12 +20,11 @@
  * a single MappedFieldType from {@link RuntimeField#asMappedFieldTypes()}
  */
 public final class LeafRuntimeField implements RuntimeField {
+    private final String name;
+    private final ToXContent toXContent;
+    private final AbstractScriptFieldType<?> mappedFieldType;
```

Comment:
Does this need to be an AbstractScriptFieldType or can it stay as a MappedFieldType?

Reply:
This change was maybe needed in some previous iteration, but it's no longer needed, I restored it.